### PR TITLE
feat: enforce pre verification gas tolerance

### PR DESCRIPTION
**Pull request by andrewwahid**
- Adds `--enforce_pre_verification_gas_tolerance` CLI argument (default: 3) that relaxes the strict `preVerificationGas` check, mirroring the existing `--enforce_gas_price_tolerance`
- Formula: `min_preVerificationGas = expected * (1 - tolerance/100)`
- Configurable via env var `VOLTAIRE_ENFORCE_PRE_VERIFICATION_GAS_TOLERANCE`
- Tolerance of 100 disables the check entirely

### CODE REVIEW

File: voltaire_bundler/gas/gas_manager.py [hunk 1]
```diff
@@ -114,18 +114,24 @@ async def verify_preverification_gas_and_verification_gas_limit(
...
-            raise ValidationException(
-                ValidationExceptionCode.InvalidFields,
-                "preVerificationGas gas is too low." +
-                f"it should be minimum : {hex(expected_preverification_gas)}",
-            )
+        min_preverification_gas = math.ceil(
+            expected_preverification_gas * (1 - (enforce_pre_verification_gas_tolerance / 100))
+        )
+
+        if enforce_pre_verification_gas_tolerance < 100:
```
Comment (Sednaoui) [inline voltaire_bundler/gas/gas_manager.py:128]: Nice addition. One edge case to tighten here: because the CLI accepts any unsigned integer, values `>= 100` turn this branch into a silent disable for preVerificationGas enforcement. I suspect you intended a bounded tolerance rather than an implicit off switch, so it would be safer to reject or clamp out-of-range values, or make disabling explicit in the interface. Could you also add tests for this feature, especially the boundary cases around `0`, `3`, `99`, and `100`?

**Review by sherifahmed990 — APPROVED** [review]
LGTM
